test/elixir_google_scraper/scraper/schemas/keyword_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule ElixirGoogleScraper.Scraper.Schemas.KeywordTest do
+  use ElixirGoogleScraper.DataCase, async: true
+
+  alias ElixirGoogleScraper.Scraper.Schemas.Keyword
+
+  describe "changeset/2" do
+    test "requires the name" do
+      attrs = %{user_id: insert(:user).id}
+      changeset = Keyword.changeset(%Keyword{}, attrs)
+
+      refute changeset.valid?
+    end
+
+    test "requires the user_id" do
+      attrs = %{name: "shopping"}
+      changeset = Keyword.changeset(%Keyword{}, attrs)
+
+      refute changeset.valid?
+    end
```

Comment:
Normally for this, we can do in one test and maybe add a success case. Some thing like this

```elixir
test "returns valid changeset if given valid attributes" do
  attrs = ...[valid attrs]...

  changeset = Keyword.changeset(%Keyword{}, attrs)

  assert changeset.valid?
end

test "returns invalid changeset if required attributes are missing" do
  changeset = Keyword.changeset(%Keyword{}, %{})

  refute changeset.valid?
  
  assert errors_on(changeset) == %{
    ...
  }
end
```

For search result, we can have another one test case for assoc constraint.

```elixir
test "returns invalid changeset if keyword does not exist" do
  attrs = ...[attributes with non existing keyword id]...

  changeset = Keyword.changeset(%Keyword{}, attrs)

  assert {:error, changeset} = Repo.insert(changeset)
  refute changeset.valid?
  assert errors_on(changeset) == %{keyword: ["does not exist"]}
end
```

For assoc constraint the error is returning after db insert, so we need to call insert and check the result after 👍